### PR TITLE
FamilyServicesTests ready

### PR DESCRIPTION
**Pull request by Aysel63**

### CODE REVIEW

File: src/main/java/az/edu/turing/business/FamilyService.java [hunk 1]
```diff
@@ -96,7 +99,11 @@ public Family getFamilyById(int index) {
 
     public Set<Pet> getPets(int familyIndex) {
         Family family = getFamilyById(familyIndex);
-        return family.getPets();
+        if (family != null) {
+            Set<Pet> pets = family.getPets();
+            return pets != null ? new HashSet<>(pets) : new HashSet<>();
```
Comment (sabuhihuseynov) [inline src/main/java/az/edu/turing/business/FamilyService.java]: I think no need checking pets' nullability here instead you can initialize pet default in Family class itself

File: src/main/java/az/edu/turing/entities/Family.java [hunk 2]
```diff
@@ -1,6 +1,13 @@
 package az.edu.turing.entities;
```
Comment (sabuhihuseynov) [inline src/main/java/az/edu/turing/entities/Family.java]: change package naming, should not be plural

File: src/test/java/az/edu/turing/FamilyServicesTests.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+import java.util.List;
+import java.util.Set;
+
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+public class FamilyServicesTests {
+    private FamilyDao familyDao;
```
Comment (sabuhihuseynov) [inline src/test/java/az/edu/turing/FamilyServicesTests.java:23]: leave a space pls

**Review by sabuhihuseynov — CHANGES_REQUESTED** [review]
pls look, change package name business to service

File: src/main/java/az/edu/turing/Main.java [hunk 4]
```diff
@@ -4,7 +4,7 @@
 import az.edu.turing.controller.FamilyController;
 import az.edu.turing.dao.FamilyDao;
 import az.edu.turing.dao.impl.CollectionFamilyDao;
-import az.edu.turing.entities.*;
+import az.edu.turing.entity.*;
```
Comment (sabuhihuseynov) [inline src/main/java/az/edu/turing/Main.java]: avoid this type of import

File: src/main/java/az/edu/turing/business/FamilyService.java [hunk 5]
```diff
@@ -1,9 +1,9 @@
 package az.edu.turing.business;
```
Comment (sabuhihuseynov) [inline src/main/java/az/edu/turing/business/FamilyService.java]: change package name to service

File: src/main/java/az/edu/turing/business/FamilyService.java [hunk 6]
```diff
@@ -51,9 +51,9 @@ public List<Family> countFamiliesWithMemberNumber(int numberOfPeople) {
                 .collect(Collectors.toList());
     }
 
-    public Family createNewFamily(Human father, Human mother) {
-        if (father == null || mother == null) throw new NullPointerException("father and mother cannot be null");
-        Family family = new Family(father, mother);
+    public Family createNewFamily(Human mother, Human father) {
+        if (mother == null || father == null) throw new NullPointerException("father and mother cannot be null");
```
Comment (sabuhihuseynov) [inline src/main/java/az/edu/turing/business/FamilyService.java]: it is not good practice throw NullPointerException like this, instead you can create your custom Exception like BadRequestException

File: src/main/java/az/edu/turing/business/FamilyService.java [hunk 7]
```diff
@@ -101,7 +101,7 @@ public Set<Pet> getPets(int familyIndex) {
         Family family = getFamilyById(familyIndex);
         if (family != null) {
             Set<Pet> pets = family.getPets();
-            return pets != null ? new HashSet<>(pets) : new HashSet<>();
+            return new HashSet<>(pets);
```
Comment (sabuhihuseynov) [inline src/main/java/az/edu/turing/business/FamilyService.java:105]: just return pets no need new HashSet<> pets already Set

**Review by sabuhihuseynov — CHANGES_REQUESTED** [review]
pls look

File: src/main/java/az/edu/turing/Main.java (hunk 4 above)
Comment (adiliso) [reply src/main/java/az/edu/turing/Main.java]: There is not other way. Beacuse we use all package fields.

File: src/main/java/az/edu/turing/business/FamilyService.java (hunk 7 above)
Comment (adiliso) [reply src/main/java/az/edu/turing/business/FamilyService.java:105]: Would it be a problem if I return a  reference directly?

File: src/main/java/az/edu/turing/Main.java (hunk 4 above)
Comment (sabuhihuseynov) [reply src/main/java/az/edu/turing/Main.java]: pls look
[How do I stop wildcard Java package imports in IntelliJ](https://stackoverflow.com/questions/26225989/how-do-i-stop-wildcard-java-package-imports-in-intellij)

**Pull request merged**